com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+                            lhsKeyCurr++;
+                        }
+                        else if (lhsKey.time > rhsKey.time)
+                        {
+                            // this is the reverse of the lhs key case
+                            //     lhsKey[curr-1].time <= rhsKey.time <= lhsKey[curr].time
+                            // so interpolate lhsKey at the rhsKey.time.
+                            lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+                            rhsKeyCurr++;
+                        }
+                        else
+                        {
+                            // should never happen, checking here to be safe
+                            Assert.IsTrue(false);
+                        }
+                    }
+                    else if (lhsValid)
+                    {
+                        // we are still processing lhsKeys, but we are out of rhsKeys, so increment lhs and evaluate rhs
+                        lhsKey = lhsCurve.keys[lhsKeyCurr];
+
+                        // rhs will be evaluated between the last rhs key and the extrapolated rhs key at the end time
+                        rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+
+                        lhsKeyCurr++;
+                    }
+                    else if (rhsValid)
+                    {
+                        // we still have rhsKeys to lerp, but we are out of lhsKeys, to increment rhs and evaluate lhs
+                        rhsKey = rhsCurve.keys[rhsKeyCurr];
+
+                        // lhs will be evaluated between the last lhs key and the extrapolated lhs key at the end time
+                        lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+
+                        rhsKeyCurr++;
+                    }
+                    else
+                    {
+                        // at least one of lhsValid or rhsValid should always be true
+                        Assert.IsTrue(false);
+                    }
+
+                    Keyframe dstKey = LerpKeyframeDirect(lhsKey, rhsKey, t);
+
+                    dstKeys[currNumKeys] = dstKey;
+                    currNumKeys++;
+                }
+
+                // if we have any keys at the same time in both curves, there will be extras in the array
+                // so trim them.
+                if (currNumKeys != dstKeys.Length)
+                {
+                    Keyframe[] trimKeys = new Keyframe[currNumKeys];
+                    for (int i = 0; i < currNumKeys; i++)
+                    {
+                        trimKeys[i] = dstKeys[i];
+                    }
+                    dstKeys = trimKeys;
+                }
+
+                m_Value = new AnimationCurve(dstKeys);
```

Comment:
As the overall improvements on the stack the components should be structs, But this garbage will be only created when you are in the middle of two volumes that override the curve, so I do not think is a big deal. Maybe @fredericv-unity3d has other options here.

Reply:
Just looked through the code again. The AnimationCurve object is instantiated, and it has native members for adding keys. So I should be able to call AnimationCurve.AddKey() to add the keys which will not create garbage since the memory is handled in native code.

Reply:
Actually, after looking at it, the m_Value is not always non-null, so I'll have to wait to see what the plan is for creating a pool of them.

Reply:
i can also suggest using `TextureCurveParameter` instead of `AnimationCurveParameter`. I believe it will make caching much easier because you won't need a pool. Then you will need a migration step to update the volumetric clouds though